src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -47,10 +50,11 @@ private class InvocationState
         public UInt160 EntryScriptHash => EntryContext?.GetState<ExecutionContextState>().ScriptHash;
         public IReadOnlyList<NotifyEventArgs> Notifications => notifications;
 
-        public ApplicationEngine(TriggerType trigger, IVerifiable container, StoreView snapshot, long gas, bool testMode = false)
+        public ApplicationEngine(TriggerType trigger, IVerifiable container, StoreView snapshot, long gas, bool testMode = false, ITraceDebugSink traceDebugSink = null)
```

Comment:
Could you create a `TraceableApplicationEngine` that inherit from this one?

Reply:
I don't understand how that would improve the code.

Reply:
You will need it for the current node execution, or you will create the instance by yourself?

Reply:
> You will need it for the current node execution, or you will create the instance by yourself?

I don't understand what you're asking. The instance of what ITraceSink or ApplicationEngine?

Reply:
When you will send `traceDebugSink`?

Reply:
OK, I see why you were confused. The traceDebugSink parameter is passed to the ApplicationEngine ctor in Blockchain.Persist, but I had missed adding it. Fixed